src/codegen/statements.rs

```diff
@@ -1181,3 +1230,25 @@ impl Type {
         }
     }
 }
+
+/// This function looks for expressions that have side effects during code execution.
+/// They must be added to the cfg event if we remove the assignment
+pub fn check_side_effects_expressions(
+    exp: &Expression,
+    ctx: &mut SideEffectsCheckParameters,
+) -> bool {
+    match &exp {
+        Expression::InternalFunctionCall { .. }
+        | Expression::ExternalFunctionCall { .. }
+        | Expression::ExternalFunctionCallRaw { .. }
+        | Expression::Constructor { .. }
+        | Expression::Assign(..)
+        | Expression::DynamicArrayPush(..)
+        | Expression::Builtin(..) => {
```

Comment:
Expression::DynamicArrayPop is missing
The Expression::Builtin can be restricted further.
You only need the builtins: PayableSend, PayableTransfer, ArrayPush, ArrayPop, Revert, Require, selfdestruct

Reply:
PayableTransfer, ArrayPush, Revert, Require and selfdestruct do not return any value, so they are not allowed inside an expression. Do I still need to take care of these cases?

Reply:
Good point, that is totally true. However the function tells you if the expression has side effects. Might as well include them with a comment that this should not occur for unused variable elimination.

Reply:
The function also adds the expressions to the CFG. Renaming it may be a better option.